### PR TITLE
Sni 3309 cookie based authentication

### PR DESCRIPTION
**Pull request by yaniv-li**

### CODE REVIEW

**Review by maoznir — APPROVED** [review]

**Comment by maoznir** [comment]
This should wait for backend support of this feature, so not merging yet

File: test/unit/videoSource.test.js [hunk 1]
```diff
@@ -201,5 +201,27 @@ describe('Raw url tests', () => {
     expect(srcs[0]).toEqual(url);
   });
 });
+describe('tests withCredentials', () => {
+  it('test withCredentials true', () => {
+    let sourceDef = {
```
Comment (e1adn) [inline test/unit/videoSource.test.js:206]: please change it const

File: test/unit/videoSource.test.js [hunk 2]
```diff
@@ -201,5 +201,27 @@ describe('Raw url tests', () => {
...
+      cloudinaryConfig: cld,
+      withCredentials: true
+    };
+    let source = new VideoSource('sea_turtle', sourceDef);
+    source.sourceTypes(['webm']);
+    let srcs = source.generateSources().map(s => s.withCredentials);
+    expect(srcs[0]).toEqual(true);
+  });
+  it('test withCredentials false', () => {
+    let sourceDef = {
```
Comment (e1adn) [inline test/unit/videoSource.test.js:216]: const

File: test/unit/videoSource.test.js [hunk 3]
```diff
@@ -201,5 +201,27 @@ describe('Raw url tests', () => {
     expect(srcs[0]).toEqual(url);
   });
 });
+describe('tests withCredentials', () => {
+  it('test withCredentials true', () => {
+    let sourceDef = {
+      cloudinaryConfig: cld,
+      withCredentials: true
+    };
+    let source = new VideoSource('sea_turtle', sourceDef);
```
Comment (e1adn) [inline test/unit/videoSource.test.js:210]: please change it const

File: test/unit/videoSource.test.js [hunk 4]
```diff
@@ -201,5 +201,27 @@ describe('Raw url tests', () => {
...
+    source.sourceTypes(['webm']);
+    let srcs = source.generateSources().map(s => s.withCredentials);
```
Comment (e1adn) [inline test/unit/videoSource.test.js:212]: same

**Pull request merged**